From 54a8a7763020395995697c9b2d3619a811313cce Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Thu, 16 Jun 2016 14:30:31 -0400
Subject: [PATCH 1809/2094] drm/amdkfd: Fix a memory leak in memory unmapping

The kmalloc memory should be freed before exiting the function.

Change-Id: If10cd8e7e09afe4b15ceb79514dea23c049b3367
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_chardev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c b/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
index 26ec03d..95c7c4a 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
@@ -1555,6 +1555,7 @@ static int kfd_ioctl_unmap_memory_from_gpu(struct file *filep,
 					mem, peer_pdd->vm);
 			radeon_flush_tlb(peer, p->pasid);
 		}
+		kfree(devices_arr);
 	} else {
 		dev->kfd2kgd->unmap_memory_to_gpu(dev->kgd, mem, pdd->vm);
 		radeon_flush_tlb(dev, p->pasid);
-- 
2.7.4

